pkg/operator/controller/ingress/deployment.go

```diff
@@ -1436,3 +1456,29 @@ func clipHAProxyTimeoutValue(val string) (string, error) {
 	}
 	return val, nil
 }
+
+// durationToHAProxyTimespec converts a time.Duration into a number that
+// HAProxy can consume, in the simplest unit possible. If the value would be
+// truncated by being converted to seconds, it outputs in milliseconds,
+// otherwise if the value wouldn't be truncated by converting to seconds, but
+// would if converted to minutes, it outputs in seconds, etc. up to a maximum
+// unit in hours (the largest unit natively supported by time.Duration).
+//
+// Also truncates values to the maximum length HAProxy allows if the value is
+// too large.
+func durationToHAProxyTimespec(duration time.Duration) string {
```

Comment:
Can we create a unit test for this to assert current functionality and so that we don't regress.

Reply:
Done